### PR TITLE
Add a bucketing event on the client side

### PR DESCRIPTION
**Pull request by nramadas**
:eyeglasses: @phil303 and @schwers and @uzi

### CODE REVIEW

**Comment by phil303** [comment]
:eyeglasses: @drunken-economist

File: src/app/featureFlags.js [hunk 1]
```diff
@@ -118,11 +119,34 @@ flags.addRule('subreddit', function (name) {
...
+      firstBuckets.add(experiment_name);
+
+      const eventTracker = getEventTracker();
+      const payload = {
+        experiment_id,
+        experiment_name,
+        variant,
+        loid: this.state.loid.loid,
+        loidcreated: this.state.loid.loidCreated,
+      };
```
Comment (phil303) [inline src/app/featureFlags.js]: seems like you're missing a bunch of data, `user_id`, `user_id36`, `user_name`, `sr_id`, and `sr_name`. Found this in https://reddit.atlassian.net/wiki/display/DA/Bucketing+Events

Comment (phil303) [inline src/app/featureFlags.js]: Also, how does `this.state` work? Is it constantly getting replaced?

**Comment by phil303** [comment]
:nail_care:

File: src/app/featureFlags.js (hunk 1 above)
Comment (nramadas) [inline src/app/featureFlags.js]: noting that we talked in person

**Comment by nramadas** [comment]
:haircut_woman:

**Comment by schwers** [comment]
:fish:

File: src/app/featureFlags.js [hunk 2]
```diff
@@ -118,11 +121,36 @@ flags.addRule('subreddit', function (name) {
...
+    // scope) to keep track of which experiments we've already bucketed.
+    if (this.state.meta.env === 'CLIENT' && !firstBuckets.has(experiment_name)) {
+      firstBuckets.add(experiment_name);
+
+      const eventTracker = getEventTracker();
+      const payload = {
+        experiment_id,
+        experiment_name,
+        variant,
+        user_id: !this.state.user.loggedOut ? parseInt(user.id, 36) : null,
```
Comment (phil303) [inline src/app/featureFlags.js]: :no_bell:

**Comment by phil303** [comment]
otherwise looks good :fish:

File: src/app/featureFlags.js [hunk 3]
```diff
@@ -118,11 +121,36 @@ flags.addRule('subreddit', function (name) {
...
+        user_name: !this.state.user.loggedOut ? user.name : null,
+        loid: this.state.user.loggedOut ? this.state.loid.loid : null,
+        loidcreated: this.state.user.loggedOut ? this.state.loid.loidCreated : null,
+      };
+
+      eventTracker.track('bucketing_events', 'bucket', omitBy(payload, isNull));
```
Comment (drunken-economist) [inline src/app/featureFlags.js]: the event type should be `cs.bucket` please

**Comment by drunken-economist** [comment]
💅 small change on the event type

File: src/app/featureFlags.js (hunk 3 above)
Comment (nramadas) [inline src/app/featureFlags.js]: whoops, thanks for catching that

**Comment by nramadas** [comment]
:haircut_woman:

**Comment by drunken-economist** [comment]
🐟  on the events

**Pull request merged**